src/lib/web-component/filter-option/filter-option-element.js

```diff
@@ -95,6 +82,29 @@ class FilterOption extends HTMLElement {
     this.toggleBtn.classList.toggle('expanded');
   }
 
+  handleCheckbox(checkboxLabel) {
+    if (this.activeCheckboxLabels.has(checkboxLabel)) {
+      this.activeCheckboxLabels.delete(checkboxLabel);
+    } else {
+      this.activeCheckboxLabels.add(checkboxLabel);
+    }
+    this.dispatchFilterChangeEvent({ isNewFilterAdded: false });
+  }
+
+  dispatchFilterChangeEvent({ isNewFilterAdded }) {
+    const filterDetail = {
+      filterObject: {
+        logKey: this.filterKey,
+        valueEquals: this.activeCheckboxLabels,
```

Comment:
Make a copy of the `activeCheckboxLabels` Set so that the internal state cannot be changed by external listeners, and that external listeners won't unexpectedly have a mutated set when the internal state changes.